pyodide_build/common.py

```diff
@@ -43,17 +43,64 @@ def xbuildenv_dirname() -> str:
 @cache
 def default_xbuildenv_path() -> Path:
     """
-    Return the default path to the cross-build environment directory.
+    Return the default path to the cross-build environment directory. This directory
+    is used when no path is provided to `pyodide xbuildenv`'s subcommands.
 
-    This directory is used when no path is provided to the `pyodide xbuildenv` subcommands.
+    The search order for determining the path is as follows:
+    1. the PYODIDE_XBUILDENV_PATH environment variable if set
+    2. the value provided by "pyodide config xbuildenv_path"
+    3. the default location based on platformdirs
+
+    Returns
+    -------
+    Path
+        The path to the cross-build environment directory
     """
+
+    # 1. check environment variable
+    env_path = os.environ.get("PYODIDE_XBUILDENV_PATH")
+    if env_path:
+        path = Path(env_path)
+        if _has_write_access(path):
+            return path
+        else:
+            logger.error(
+                "The directory specified in PYODIDE_XBUILDENV_PATH (%s) is not writable. "
+                "Falling back to other locations.",
+                path,
+            )
+
+    # 2. check "pyodide config xbuildenv_path"
+    try:
+        from pyodide_build.config import ConfigManager
```

Comment:
You can use [`get_host_build_flag("PYODIDE_XBUILDENV_PATH")`](https://github.com/pyodide/pyodide-build/blob/c83f0ae3549f206fcefe02b161409d11f28f6de0/pyodide_build/build_env.py#L166-L174) instead of instantiating the ConfigManager.

Also, the env variable [takes dominance](https://github.com/pyodide/pyodide-build/blob/c83f0ae3549f206fcefe02b161409d11f28f6de0/pyodide_build/config.py#L26-L32), so I think you don't need (1).

Reply:
Thanks for the pointer, I didn't know about this! It's much simpler, and I checked – the environment variable does dominate. I addressed this in eefc20c254a4e74adcf4faf9f2af5c9ae45c48b6.